homeassistant/components/tado/config_flow.py

```diff
@@ -110,6 +113,53 @@ async def async_step_homekit(
         self._abort_if_unique_id_configured()
         return await self.async_step_user()
 
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        _LOGGER.debug("Importing Tado from configuration.yaml")
+        username = import_config[CONF_USERNAME]
+        password = import_config[CONF_PASSWORD]
+        home_id = import_config.get(CONF_HOME_ID, None)
+
+        self._async_abort_entries_match(
+            {
+                CONF_USERNAME: username,
+                CONF_PASSWORD: password,
+                CONF_HOME_ID: home_id,
+            }
+        )
+
+        try:
+            await validate_input(
+                self.hass,
+                {
+                    CONF_USERNAME: username,
+                    CONF_PASSWORD: password,
+                },
+            )
+        except RuntimeError:
+            _LOGGER.exception("Unexpected exception while importing Tado")
+            async_create_issue(
```

Comment:
We should abort here. Probably create the issue in the platform `async_get_scanner` instead depending on the result of the flow.